cmake/sketch_preprocess_sources.cmake

```diff
@@ -27,7 +27,7 @@ function(sketch_preprocess_sources)
         COMPILE_OPTIONS "-include;Arduino.h;-include;${SRCFILE}.h"
         OBJECT_DEPENDS "${SRCFILE}.h"
       )
-      list(APPEND SRCLIST ${SRCFILE}.cpp)
+      list(APPEND SRCLIST ${CMAKE_CURRENT_BINARY_DIR}/${SRC_BASE_NAME}.cpp)
```

Comment:
Can you apply the same change on line 25?
```py
set_source_files_properties(${SRCFILE}.cpp
```

Reply:
Seems I missed some other places where ${CMAKE_CURRENT_BINARY_DIR} was not prepended to the source file. I have pushed another commit. This has uncovered another issue with my code however as the generated include file has definitions like `bool help_cmd(CommandParser *)` which trip the compiler up as the generated `sketch.ino.h` gets added  included in `sketch.ino.cpp` before the actual includes for the library. I will see how currently Arduino build manages the includes and get back with a fix.

Reply:
Actually this appears to be a "known" issue with how Arduino ino file pre-processing works and it not specific to the cmake support. 
See [https://www.gammon.com.au/forum/?id=12625]. I will find a solution to this in my own code.